Services/Administration/OJS.Services.Administration.Business/Tests/TestsBusinessService.cs

```diff
@@ -0,0 +1,190 @@
+﻿namespace OJS.Services.Administration.Business.Tests;
+
+using Microsoft.EntityFrameworkCore;
+using OJS.Common;
+using OJS.Common.Helpers;
+using OJS.Data.Models.Tests;
+using OJS.Servers.Administration.Models.Tests;
+using OJS.Services.Administration.Business.Problems;
+using OJS.Services.Administration.Data;
+using OJS.Services.Administration.Models.Tests;
+using OJS.Services.Common;
+using OJS.Services.Common.Models;
+using OJS.Services.Infrastructure.Exceptions;
+using SoftUni.AutoMapper.Infrastructure.Extensions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Transactions;
+
+public class TestsBusinessService : AdministrationOperationService<Test, int, TestAdministrationModel>, ITestsBusinessService
+{
+    private readonly ITestsDataService testsDataService;
+    private readonly ITestRunsDataService testRunsDataService;
+    private readonly IProblemsDataService problemsDataService;
+    private readonly IFileSystemService fileSystemService;
+    private readonly IZippedTestsParserService zippedTestsParserService;
+    private readonly ISubmissionsDataService submissionsDataService;
+    private readonly IProblemsBusinessService problemsBusinessService;
+    private readonly IZipArchivesService zipArchivesService;
+
+    public TestsBusinessService(
+        ITestsDataService testsDataService,
+        ITestRunsDataService testRunsDataService,
+        IProblemsDataService problemsDataService,
+        IFileSystemService fileSystemService,
+        IZippedTestsParserService zippedTestsParserService,
+        ISubmissionsDataService submissionsDataService,
+        IProblemsBusinessService problemsBusinessService,
+        IZipArchivesService zipArchivesService)
+    {
+        this.testsDataService = testsDataService;
+        this.testRunsDataService = testRunsDataService;
+        this.problemsDataService = problemsDataService;
+        this.fileSystemService = fileSystemService;
+        this.zippedTestsParserService = zippedTestsParserService;
+        this.submissionsDataService = submissionsDataService;
+        this.problemsBusinessService = problemsBusinessService;
+        this.zipArchivesService = zipArchivesService;
+    }
+
+    public override async Task<TestAdministrationModel> Get(int id)
+    {
+        var test = await this.testsDataService.GetByIdQuery(id)
+            .Include(t => t.Problem)
+            .FirstOrDefaultAsync();
```

Comment:
Use .MapCollection here, instead of doing .Map on a single entity from the db. This way you will not need `Include` clause, scales better and is arguably more performant.

Reply:
Fixed.